f5/bigip/resource.py

```diff
@@ -369,6 +377,48 @@ def raw(self):
         """
         return self.__dict__
 
+    def _key_dot_replace(self, rdict):
+        """Replace fullstops in returned keynames"""
+        temp_dict = {}
+        for key, value in rdict.items():
+            if isinstance(value, dict):
+                value = self._key_dot_replace(value)
+            temp_dict[key.replace('.', '_')] = value
+        return temp_dict
+
+    def _get_nest_stats(self, rdict):
+        """Helper method to deal with nestedStats
+
+        as json format changed in v12.x
+        """
+        for x in rdict:
+            check = urlparse.urlparse(x)
+            if check.scheme:
+                nested_dict = rdict[x]['nestedStats']
+                tmp_dict = nested_dict['entries']
+                return self._key_dot_replace(tmp_dict)
+
+        return self._key_dot_replace(rdict)
+
+    def _get_stats_raw(self):
+        """Displays JSON object transformed into python dictionary"""
+        read_session = self._meta_data['bigip']._meta_data['icr_session']
+        base_uri = self._meta_data['container']._meta_data['uri'] + 'stats/'
+        response = read_session.get(base_uri)
```

Comment:
I was looking at it but stats work a bit differently, apart from the need to append URI with 'stats/' refresh grabs the json and then performs local_update: 

https://github.com/zancas/f5-common-python/blob/statsmodule/f5/bigip/resource.py#L558

We do not process stats json the same way, if you look at the update_stats helper method in PathElement you probably know what i mean: 

https://github.com/zancas/f5-common-python/blob/statsmodule/f5/bigip/resource.py#L416

Given the fact as @caphrim007 noticed that we might run into situation where the json structure is different, we would need to add/tweak that _update_stats method to fit the situation and cover the inconsistent output (great programming there i must say ;)).

That is why i think we need to evolve this once we start adding more and more resources.